src/app/components/addFinancial/addFinancial.component.ts

```diff
@@ -18,23 +20,43 @@ export class AddFinancialComponent implements OnInit {
                 libelle: [
                     '', Validators.required
                 ],
-                rentree: [
-                    '', Validators.required
+                inOrOut: [
+                    '1', Validators.required
                 ],
-                sortie: ['', Validators.required]
+                amount: [, Validators.required]
             });
     }
 
     ngOnInit() {}
 
     add() {
+        let data 
+        if (this.form.value.inOrOut === '1') {
+            data = {
+                date: this.form.value.date,
+                libelle: this.form.value.libelle,
+                rentree: +this.form.value.amount,
+            };
+        }else{
+             data = {
+                date: this.form.value.date,
+                libelle: this.form.value.libelle,
+                sortie:  +this.form.value.amount,
```

Comment:
what is this  `+` ?

Reply:
'+' cast string to number, it's because of JsonLd that claim me number for this attribute